internal/command/container.go

```diff
@@ -0,0 +1,393 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fyne-io/fyne-cross/internal/icon"
+	"github.com/fyne-io/fyne-cross/internal/log"
+	"github.com/fyne-io/fyne-cross/internal/volume"
+	"golang.org/x/sys/execabs"
+)
+
+const (
+	// fyneBin is the path of the fyne binary into the docker image
+	fyneBin = "/usr/local/bin/fyne"
+	// gowindresBin is the path of the gowindres binary into the docker image
+	gowindresBin = "/usr/local/bin/gowindres"
+)
+
+type ContainerRunner interface {
+	NewImageContainer(arch Architecture, OS string, image string) ContainerImage
+
+	Debug(v ...interface{})
+	GetDebug() bool
+}
+
+type AllContainerRunner struct {
+	ContainerRunner
+
+	Env  map[string]string // Env is the list of custom env variable to set. Specified as "KEY=VALUE"
+	Tags []string          // Tags defines the tags to use
+
+	vol volume.Volume
+
+	debug bool
+}
+
+type ContainerImage interface {
```

Comment:
As said above not sure about the name here. Maybe something related to a builder ?

Reply:
What about `ImageInstance`?

Reply:
I think there is a peculiar relationship between `ContainerRunner`, `AllContainerRunner` and `ContainerImage` here, solving the overall model might lead to a good name here.
What is the intent of these interfaces/structs?

There is also a circular reference between `ContainerRunner` and `ContainerImage`. I think that this discussion is not so much a question of naming as the grouping itself.

Reply:
A `ContainerImage` define the os/architecture and link to the proper engine logic to build for.

`AllContainerRunner` is the common logic between all engine.
`AllContainerImage` is the common property/logic between all image.

I was expecting to have to replace some of those when doing the kubernetes engine, but in fact they are completely generic.

Reply:
It sounds like the `AllContainerRunner` could then be `baseEngine` (not-exported base type for common code) and then the `AllContainerImage` could be `baseContainer` or `baseImage` or whatever the right name  is and the implementations could embed that the same way?